crates/tasks/src/lib.rs

```diff
@@ -177,14 +179,41 @@ impl Future for TaskManager {
 
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         let err = ready!(self.get_mut().panicked_tasks_rx.poll_recv(cx));
-        Poll::Ready(err.map(PanickedTaskError).expect("stream can not end"))
+        Poll::Ready(err.expect("stream can not end"))
     }
 }
 
-/// Error with the name of the task that panicked.
+/// Error with the name of the task that panicked and an error downcasted to string, if possible.
 #[derive(Debug, thiserror::Error)]
-#[error("Critical task panicked: `{0}`")]
-pub struct PanickedTaskError(&'static str);
+pub struct PanickedTaskError {
+    task_name: &'static str,
+    error: Option<String>,
+}
+
+impl Display for PanickedTaskError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let task_name = self.task_name;
+        if let Some(error) = &self.error {
+            write!(f, "Critical task `{task_name}` panicked: `{error}`")
+        } else {
+            write!(f, "Critical task `{task_name}` panicked")
```

Comment:
nit:
use shared `write!(f, "Critical task `{task_name}` panicked")` always and `write!(f, ": `{error}`")` in if let

Reply:
yeah, I wrote it like that at first but didn't like the dangling `:` in the second write

Reply:
I guess this doesn't make much difference